### PR TITLE
EbayFakeMenuButton component

### PR DESCRIPTION
**Pull request by anch-0**
Porting over from our internal repository.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -65,6 +65,7 @@
             "named": "never",
             "asyncArrow": "always"
         }],
+        "array-bracket-spacing": "off",
```
Comment (abiyasa) [inline .eslintrc:68]: Just like previous PR https://github.com/eBay/ebayui-core-react/pull/146/files#r1018759202, better to have this on separated PR.

File: src/ebay-menu-button/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# ebay-menu-button
+
+## Demo
+[Storybook](https://pages.github.corp.ebay.com/React/ebayui-core-react/master/?path=/story/ebay-menu-button--default)
+
+## Install
+```
```
Comment (abiyasa) [inline src/ebay-menu-button/README.md:7]: What are the differences between this PR and the previous PR (https://github.com/eBay/ebayui-core-react/pull/146)? Both PR contains the same and overlapping changes on ebay-menu-button, the PR description is not clear enough.

If so, that could cause merging conflict as well. I would suggest to split this PR into smaller PRs (1 PR for each component). Smaller PR is more focus, smaller risk to deploy, and easier to review/merge.

File: src/common/component-utils/utils.ts [hunk 3]
```diff
@@ -10,9 +10,10 @@ export function excludeComponent(nodes: ReactNode = [], componentType: FC): Reac
     return elements.filter(({ type }) => type !== componentType)
 }
 
-export function filterByType(nodes: ReactNode = [], componentType: FC, fn?: () => void): ReactElement[] {
+export function filterByType(nodes: ReactNode = [], componentType: FC | FC[], fn?: () => void): ReactElement[] {
     const elements = Children.toArray(nodes) as ReactElement[]
-    return elements.filter(({ type }) => type === componentType)
+    const types = [componentType].flat()
+    return elements.filter(({ type }) => types.includes(type as any))
```
Comment (abiyasa) [inline src/common/component-utils/utils.ts:16]: This is the same changes on the other PR (https://github.com/eBay/ebayui-core-react/pull/146). See my previous comment about splitting into smaller PRs.

File: tsconfig.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
         "module": "commonjs",
         "lib": [
             "dom",
-            "es2017",
+            "esnext",
```
Comment (abiyasa) [inline tsconfig.json:7]: Same as my comment on eslint rule change, split into separated PR.

**Review by abiyasa — CHANGES_REQUESTED** [review]
It seems there are some overlapping with the other PR (https://github.com/eBay/ebayui-core-react/pull/146). As suggested on my comment, make more sense to split this PR into smaller ones, one PR per component since those components can be implemented/delivered independently.

File: src/ebay-menu-button/README.md (hunk 2 above)
Comment (anch-0) [reply src/ebay-menu-button/README.md:7]: I ported over these changes from @darkwebdev's original PR so I will let him take the lead on this one

**Comment by darkwebdev** [comment]
deprecated in favor of #150

**Pull request closed without merging**